spec/models/measurement_unit_spec.rb

```diff
@@ -19,56 +17,60 @@
-    let(:units) { described_class.send(:measurement_units) }
-
-    it "will return the hash of all measurement units" do
-      expect(units).to be_instance_of Hash
+  describe '.measurement_unit' do
+    let(:measurement_unit) do
+      create(
+        :measurement_unit,
+        :with_description,
+        measurement_unit_code: measurement_unit_code,
+      )
     end
 
-    it "will include individual units indexed by key" do
-      expect(units).to include("ASV")
-    end
-  end
+    let(:measurement_unit_code) { 'ASV' }
 
-  describe '.measurement_unit' do
     context 'with valid measurement unit' do
-      it { expect(MeasurementUnit.measurement_unit('ASV')).to include('unit' => 'percent') }
+      let(:unit_code) { 'ASV' }
+      let(:unit_key) { 'ASV' }
+
+      it { expect(described_class.measurement_unit(unit_code, unit_key)).to include('unit' => 'percent') }
     end
 
     context 'with missing measurement unit present in database' do
-      before { allow(Raven).to receive(:capture_message).and_return(true) }
-      before { allow(MeasurementUnit).to receive(:measurement_units).and_return({}) }
+      subject(:result) { described_class.measurement_unit(unit_code, unit_key) }
 
-      subject! { MeasurementUnit.measurement_unit(unit_code) }
+      before do
+        measurement_unit
+        allow(Raven).to receive(:capture_message).and_return(true)
+        allow(described_class).to receive(:measurement_units).and_return({})
+      end
 
-      let(:unit_code) { measurement_unit.measurement_unit_code }
+      let(:unit_code) { 'ASV' }
+      let(:unit_key) { 'ASVX' }
       let(:unit_description) { measurement_unit.description }
 
       it { is_expected.to include('measurement_unit_code' => unit_code) }
+      it { is_expected.to include('measurement_unit_qualifier_code' => 'X') }
       it { is_expected.to include('unit' => nil) }
-      it { is_expected.to include('abbreviation' => '') }
+      it { is_expected.to include('abbreviation' => measurement_unit.abbreviation) }
       it { is_expected.to include('unit_question' => "Please enter unit: #{unit_description}") }
-      it { is_expected.to include('unit_hint' => nil) }
-      it { expect(Raven).to have_received(:capture_message) }
+      it { is_expected.to include('unit_hint' =>  "Please correctly enter unit: #{unit_description}") }
+
+      it 'sends a message to Sentry' do
+        result
+        expect(Raven).to have_received(:capture_message)
+      end
     end
 
     context 'with measurement unit not in the database' do
-      let(:unit) { MeasurementUnit.measurement_unit('UNKNOWN') }
+      let(:unit) { described_class.measurement_unit('UNKNOWN', 'UNKNOWN') }
 
-      it "will raise an InvalidMeasurementUnit exception" do
+      it 'will raise an InvalidMeasurementUnit exception' do
         expect { unit }.to raise_exception(MeasurementUnit::InvalidMeasurementUnit)
       end
     end
-
-    context 'without specifying a unit' do
```

Comment:
Removed as this seemed defensive since we're the clients and the writers of the code that drives this and we'd know this error was propagated in the service if we weren't calling it with an argument